src/articulation.ts

```diff
@@ -143,13 +160,24 @@ const getInitialOffset = (note, position) => {
       return 0;
     }
   }
-};
+}
 
 export class Articulation extends Modifier {
-  static get CATEGORY() {
+  note?: Note;
+  readonly type: string;
+
+  protected render_options: { font_scale: number };
+  protected articulation?: ArticulationStruct;
+  // glyph defined calling reset in constructor
+  // eslint-disable-next-line
+  protected glyph!: Glyph;
```

Comment:
Why `!` here?

Reply:
Because I can tell the compiler that it is surely defined in the construtor. I tried to explain it in the comment above. The `constructor` calls `reset` which defines `this.glyph`

Reply:
@0xfe would you like a different approach? `!` is thought to tell the compiler: "forget it I know better"

Reply:
Ah, okay I see.